src/fetchTokens.js

```diff
@@ -0,0 +1,98 @@
+import axios from 'axios';
+import icrc1IDL from './ic/idl/icrc1.did.js';
+import { Actor, HttpAgent } from '@dfinity/agent';
+import fetch, { Headers } from 'node-fetch';
+global.fetch = fetch;
+global.Headers = Headers;
+import dotenv from 'dotenv';
+import { loadJSON, updateTokenListJson } from './helpers.js';
+import { HEADERS_CONFIG } from './contants.js';
+dotenv.config();
+
+const tokenFilePath =
+  process.env.IC_ENVIRON === 'testnet'
+    ? './tokenlist.testnet.json'
+    : './tokenlist.json';
+
+const TokensJson = loadJSON(tokenFilePath);
+
+const updateTokenList = (tokens) => {
+  tokens.forEach((token) => {
+    const foundIndex = TokensJson.tokens.findIndex((i) => i.id === token.id);
```

Comment:
Using find makes more sense. then you can get the current token item itself, instead of accessing every time via index